### DISCUSSION TITLE
Range slider

### DISCUSSION THREAD

**Discussion by granthamtaylor** [Textualize/textual#3592] [Ideas]
I am very interested in using Textual to create a TUI for my AutoML project.

A common theme with model training pipelines is a large set of hyper parameters with seemingly arbitrary ranges. 

I feel like a range slider widget would be an excellent way to provide context on recommended ranges for numerical inputs.

Ideally, the range would have formatting support, provide a tooltip when being updated, may be interacted with via both mouse and keyboard, and may provide custom range distributions (log scale, discrete, custom labels, etc).

I am curious if anyone else is interested in the idea!

**Comment by davep** [comment]
You may find [this discussion](https://github.com/Textualize/textual/discussions/2824) interesting; @TomJGooding created [textual-slider](https://github.com/TomJGooding/textual-slider) which would seem to be what you want, or perhaps very close to what you want?

**Reply by granthamtaylor** [reply]
Yes this is very close to what I am searching for! Thank you.
